src/main/java/org/medicmobile/webapp/mobile/AppNotificationManager.java

```diff
@@ -0,0 +1,200 @@
+					TimeUnit.MINUTES
+			)
+					.addTag(NotificationWorker.NOTIFICATION_WORK_REQUEST_TAG)
+					.build();
+
+			WorkManager.getInstance(context).enqueueUniquePeriodicWork(
+					NotificationWorker.NOTIFICATION_WORK_NAME,
+					ExistingPeriodicWorkPolicy.KEEP,
+					request
+			);
+			log(context, "startNotificationWorker() :: Started Notification Work Manager...");
+		}
+	}
+
+	public void stopNotificationWorker() {
+		appDataStore.saveString(TASK_NOTIFICATIONS_KEY, "[]");
+		WorkManager.getInstance(context).cancelAllWorkByTag(NotificationWorker.NOTIFICATION_WORK_REQUEST_TAG);
+		log(context, "stopNotificationWorker() :: Stopped notification work manager");
+	}
+
+	void cancelAllNotifications() {
+		manager.cancelAll();
+	}
+
+	/**
+	 * @param jsArrayString string notifications sorted by readyAt in descending order
+	 * @throws JSONException throws JSONException
+	 * Method is blocking don't run on UI thread
+	 */
+	public void showNotificationsFromJsArray(String jsArrayString) throws JSONException {
+		JSONArray dataArray = Utils.parseJSArrayData(jsArrayString);
+		if (dataArray.length() == 0) {
+			return;
+		}
+		showMultipleTaskNotifications(dataArray);
+	}
+
+	private void showMultipleTaskNotifications(JSONArray dataArray) throws JSONException {
+		Intent intent = new Intent(context, EmbeddedBrowserActivity.class);
+		intent.setData(Uri.parse(TextUtils.concat(appUrl, "/#/tasks").toString()));
+		long maxNotifications = appDataStore.getLongBlocking(MAX_NOTIFICATIONS_TO_SHOW_KEY, 8);
+		long latestStoredTimestamp = getLatestStoredTimestamp(getStartOfDay());
+		int counter = 0;
+		for (int i = 0; i < dataArray.length(); i++) {
+			JSONObject notification = dataArray.getJSONObject(i);
+			long readyAt = notification.getLong("readyAt");
+			long endDate = notification.getLong("endDate");
+			long dueDate = notification.getLong("dueDate");
+			if (isValidNotification(latestStoredTimestamp, readyAt, dueDate, endDate)) {
+				int notificationId = notification.getString("_id").hashCode();
+				String contentText = notification.getString("contentText");
+				String title = notification.getString("title");
+				showNotification(intent, notificationId, title, contentText);
+				counter++;
+			}
+			if (counter >= maxNotifications) {
+				break;
+			}
+		}
+		saveLatestNotificationTimestamp(dataArray.getJSONObject(0).getLong("readyAt"));
+	}
+
+	private boolean isValidNotification(long latestStoredTimestamp, long readyAt, long dueDate, long endDate) {
+		long startOfDay = getStartOfDay();
+		return readyAt > latestStoredTimestamp && dueDate <= startOfDay && endDate >= startOfDay;
+	}
+
+	public void saveLatestNotificationTimestamp(long value) {
+		appDataStore.saveLong(LATEST_NOTIFICATION_TIMESTAMP_KEY, value);
+	}
+
+	public long getStartOfDay() {
+		return LocalDate.now()
+				.atStartOfDay(ZoneId.systemDefault())
+				.toInstant().toEpochMilli();
+	}
+
+	private long getLatestStoredTimestamp(long startOfDay) {
+		if (isNewDay(startOfDay)) {
```

Comment:
I may be missing something here, but why do we need this `isNewDay` check here?  Seems like the logic in `isValidNotification` should work if we always just return the stored value for `LATEST_NOTIFICATION_TIMESTAMP_KEY` (or `0` if we don't have a value.)